packages/shell-api/src/database.ts

```diff
@@ -617,6 +618,21 @@ export default class Database extends ShellApiWithMongoClass {
     }
   }
 
+  @returnsPromise
+  @apiVersions([])
+  async createEncryptedCollection(name: string, options: CreateEncryptedCollectionOptions): Promise<{ collection: Collection, encryptedFields: Document }> {
+    assertArgsDefinedType([name], ['string'], 'ClientEncryption.createEncryptedCollection');
+    assertArgsDefinedType([options], ['object'], 'ClientEncryption.createEncryptedCollection');
+    assertKeysDefined(options, ['provider', 'createCollectionOptions']);
```

Comment:
Maybe asserting args are fine but am I being too itchy going in and asserting keys? I have a feeling that it would be fine to leave it to libmongocrypt and throw when required params are missing.

Reply:
This seems fine either way to me. I don’t think you really need to validate here, though, since you’re already validating arguments for the `ClientEncryption` method that this wraps?